lib/exabgp/reactor/network/outgoing.py

```diff
@@ -21,22 +21,24 @@ def __init__ (self, afi, peer, local, port=179,md5='',ttl=None):
 
 		self.logger.wire("attempting connection to %s:%d" % (self.peer,port))
 
-		self.peer = peer
 		self.ttl = ttl
 		self.afi = afi
 		self.md5 = md5
 		self.port = port
 
 		try:
 			self.io = create(afi)
-			MD5(self.io,peer,port,md5)
+			MD5(self.io,self.peer,port,md5)
 			if afi == AFI.ipv4:
-				TTL(self.io, peer, self.ttl)
+				TTL(self.io, self.peer, self.ttl)
 			elif afi == AFI.ipv6:
-				TTLv6(self.io, peer, self.ttl)
-			bind(self.io,local,afi)
-			async(self.io,peer)
-			connect(self.io,peer,port,afi,md5)
+				TTLv6(self.io, self.peer, self.ttl)
+			if local:
+				bind(self.io,self.local,afi)
+			async(self.io,self.peer)
+			connect(self.io,self.peer,port,afi,md5)
+			if not self.local:
+				self.local = self.io.getsockname()[0]
```

Comment:
Not quite sure what you mean here. Do you mean update the local_address field of neighbor with the auto discovered address and then have a separate field that we can query later for display purposes to tell us if it was auto-discovered or not?